src/main/java/io/github/eocqrs/kafka/fake/ImmutableReentrantLock.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2022 Aliaksei Bialiauski, EO-CQRS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.eocqrs.kafka.fake;
+
+import java.io.Serial;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Immutable Lock.
+ *
+ * @author Aliaksei Bialiauski (abialiauski.dev@gmail.com)
+ * @since 0.2.3
+ */
+final class ImmutableReentrantLock extends ReentrantLock {
```

Comment:
why is it necessary?

Reply:
@h1alexbel it's necessary for resolving resource (in this case it's file) locking issues with concurrent resource access

WDYT?

Reply:
@h1alexbel Yes, but why is only `ReentrantReadWriteLock` bad? For what reason do we need a new one?

Reply:
@l3r8yJ we need it, simply because it never changes, since it has declared `serialVersionUID`